launcher/LaunchController.cpp

```diff
@@ -220,13 +219,34 @@ void LaunchController::login()
         m_session->demo = m_demo;
         m_accountToUse->fillSession(m_session);
 
-        // Launch immediately in true offline mode
-        if (m_accountToUse->accountType() == AccountType::Offline) {
-            launchInstance();
+        MinecraftAccountPtr accountToCheck;
+
+        if (m_accountToUse->ownsMinecraft())
+            accountToCheck = m_accountToUse;
+        else if (const MinecraftAccountPtr defaultAccount = APPLICATION->accounts()->defaultAccount();
+                 defaultAccount != nullptr && defaultAccount->ownsMinecraft()) {
+            accountToCheck = defaultAccount;
+        } else {
+            for (int i = 0; i < APPLICATION->accounts()->count(); i++) {
+                MinecraftAccountPtr account = APPLICATION->accounts()->at(i);
+                if (account->ownsMinecraft())
+                    accountToCheck = account;
+            }
+        }
+
+        if (accountToCheck == nullptr) {
+            if (!m_session->demo)
+                m_session->demo = askPlayDemo();
+
+            if (m_session->demo)
+                launchInstance();
+            else
+                emitFailed(tr("Launch cancelled - account does not own Minecraft."));
+
             return;
         }
 
-        switch (m_accountToUse->accountState()) {
+        switch (accountToCheck->accountState()) {
```

Comment:
so we are gonna try to launch online with an offline account?

Reply:
Yep it does launch the instance in `Launched instance in online mode`

Reply:
Just make sure that if the m_accountToUse is offline then `  m_session->wants_online = false;`

Reply:
ooops